modules/core/src/v2/coins/algo.ts

```diff
@@ -205,15 +304,29 @@ export class Algo extends BaseCoin {
     const pair = generateAccountFromSeed(seed);
     const sk = pair.sk;
 
+    // decode our tx
+    let transaction;
+    try {
+      const txToHex = Buffer.from(txBase64, 'base64');
+      const decodedTx = Encoding.decode(txToHex);
```

Comment:
In `explainTransaction` you do almost the same, but check for `const txnForDecoding = decodedTx.txn || decodedTx;`

Why not here? And if you do, please move the code to its own method.

Reply:
the difference is explainTransaction can take in an unsigned or signed transaction, whereas signTransaction is only primed for an unsigned tx. we would have to call merge on this if we expect more than 1 sig (which we don't so far)